blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala

```diff
@@ -200,10 +201,8 @@ private final class Http1Connection[F[_]](
           if (userAgent.nonEmpty && !req.headers.contains[`User-Agent`])
             rr << userAgent.get << "\r\n"
 
-          val mustClose: Boolean = req.headers.get[HConnection] match {
-            case Some(conn) => checkCloseConnection(conn, rr)
-            case None => getHttpMinor(req) == 0
-          }
+          val mustClose: Boolean =
+            checkRequestCloseConnection(req.headers.get[HConnection], getHttpMinor(req), NullWriter)
```

Comment:
I had a hard time teasing apart the side effect of writing the `Connection: close` header, but we don't even want the side effect on the client side.

Alternatives would be to:
- duplicate the connection persistence logic
- put a protected method in `Http1Stage` to represent the side effect, or lack thereof
- wrap this writer in an Option
- question all our lives' choices